src/NServiceBus.Callbacks/ExtendableOptionsExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace NServiceBus
+{
+    using System.Threading;
+    using NServiceBus.Extensibility;
+
+    static class ExtendableOptionsExtensions
+    {
+        public static void RegisterTokenInternal(this ExtendableOptions options, CancellationToken cancellationToken)
+        {
+            var extensions = options.GetExtensions();
+            UpdateRequestResponseCorrelationTableBehavior.RequestResponseParameters data;
```

Comment:
+1



*From:* Szymon Pobiega [mailto:notifications@github.com]
*Sent:* Donnerstag, 7. Mai 2015 08:25
*To:* Particular/NServiceBus
*Cc:* Daniel Marbach
*Subject:* Re: [NServiceBus] [WIP] Callback cancelable (#2730)



In src/NServiceBus.Callbacks/ExtendableOptionsExtensions.cs
<https://github.com/Particular/NServiceBus/pull/2730#discussion_r29827382>:

> @@ -0,0 +1,23 @@

> +﻿namespace NServiceBus

> +{

> +    using System.Threading;

> +    using NServiceBus.Extensibility;

> +

> +    static class ExtendableOptionsExtensions

> +    {

> +        public static void RegisterTokenInternal(this ExtendableOptions options, CancellationToken cancellationToken)

> +        {

> +            var extensions = options.GetExtensions();

> +            UpdateRequestResponseCorrelationTableBehavior.RequestResponseParameters data;

Not regarding this PR, but in general I think we need to provide GetOrCreate
convenience method on the extensions object because we will have this kind
of code everywhere.

—
Reply to this email directly or view it on GitHub
<https://github.com/Particular/NServiceBus/pull/2730/files#r29827382>.